### PR TITLE
[LDAP] expose isBound method of ConnectionInterface implementation in Ldap

### PR DESCRIPTION
**Pull request by fruitwasp**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Ldap/LdapInterface.php [hunk 1]
```diff
@@ -52,6 +53,11 @@ public function query($dn, $query, array $options = array());
     public function getEntryManager();
 
     /**
+     * @return ConnectionInterface
+     */
+    public function getConnection();
```
Comment (xabbuh) [inline src/Symfony/Component/Ldap/LdapInterface.php]: This is a BC break and cannot be done.

**Review by xabbuh — CHANGES_REQUESTED** [review]

**Comment by xabbuh** [comment]
Can you explain the use case for this change?

**Comment by fruitwasp** [comment]
I would like to access [Connection.isBound](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Ldap/Adapter/ExtLdap/Connection.php#L39) to check whether it's already bound or not and therefore should bind to ldap or not.

**Comment by ro0NL** [comment]
Makes sense to have `LdapInterface::isBound` along with `bind`. I dont think it should expose the whole connection object.

And apparently it can be done :) see https://github.com/symfony/symfony/issues/20258#issuecomment-255152599

**Comment by fruitwasp** [comment]
As adding methods to the `LdapInterface` breaks bc, I've only added the method `isBound` to the implementation `Ldap`, which works exactly like `Ldap::bind`

**Comment by fruitwasp** [comment]
@csarrazi

**Comment by csarrazi** [comment]
The thing is, I am not really fond of exposing the `isBound()` method. The main reason for this is that you won't be able to tell which DN was used when binding to the LDAP server, which may induce someone into thinking that a user is logged in, whereas in reality, you were bound using a search user, or even bound anonymously.

At some point, I was actually even considering removing the `isBound()` method and the associated property from the Connection object as it may induce people into thinking that they are already collected with the correct user.

**Comment by csarrazi** [comment]
Another addition. The only reason why the `isBound()` method and its associated property exist is to prevent re-binding with an anonymous user when issuing a query, if already bound previously.
If there was no bound user before, then the query issues an anonymous bind.

To sum up, it is not the responsibility of the ldap component to track which user is bound to the current connection, but the application.

All in all, I'm 👎 on exposing the `isBound()` method.

**Comment by fabpot** [comment]
Closing then

**Pull request closed without merging**